File: ass3/pr4.c
#include<stdio.h>
int main()
{
	int n,m;
	printf("Enter the row and column number");
	printf("\n");
	scanf("%d",&n);
	
	int a[n][n];
	printf("Enter the elemnts in the row");
	printf("\n");
	int i,j;
	for(i=0;i<n;i++)
	{
		for(j=0;j<n;j++)
		{
			scanf("%d",&a[i][j]);
		}
	}
	for(i=0;i<n;i++)
	{
		for(j=0;j<n;j++)
		{
			printf("%d",a[i][j]);
		}
		printf("\n");
	}
	int tot=n*n;
	int zero=0;
	for(i=0;i<n;i++)
	{
		for(j=0;j<n;j++)
		{
			if(a[i][j]==0)
			{
				zero=zero+1;
			}
		}
	}
	
	if(zero>((n*n)/2))
	{
		printf("Sparse matrix");
	}
	else
	{
		printf("not sparse matrix");
	}
	
}

File: Ass3/pr4.py
n=int(input("Enter the size of the matrix:"))
arr=[]
for i in range(0,n):
    l=[]
    for j in range(0,n):
        num=int(input("Enter the elements:"))
        l.append(num)
    arr.append(l)
z=0
for i in range(0,n):
    for j in range(0,n):
        print(arr[i][j]," ",end="")
    print("\n")
for i in range(0,n):
    for j in range(0,n):
        if(arr[i][j]==0):
            z+=1
if(z>((n*n)//2)):
    print("Sparse")
else:
    print("not sparse")
